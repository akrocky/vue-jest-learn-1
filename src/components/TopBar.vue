<template>
  <div>
    <v-app-bar
      dark
    >

      <v-toolbar-title class="logo-text">
        Regex Breakfast
      </v-toolbar-title>

      <v-spacer></v-spacer>

      <v-card outlined rounded>
        <v-card-text class="white--text pa-2">
          <v-progress-linear
            :value="$store.getters.percentDone"
            height="10"
            color="#ec1a25"
          ></v-progress-linear>
          Challenge #{{ $store.state.currentIndex + 1 }}
          out of {{ $store.state.challenges.length }}
        </v-card-text>
      </v-card>

      <v-spacer></v-spacer>

      <v-btn outlined tile @click="drawer = true">
        Challenges
      </v-btn>
    </v-app-bar>

    <v-navigation-drawer
      v-model="drawer"
      overlay-opacity="0"
      absolute
      temporary
      right
    >
      <v-list
        nav
        dense
      >
        <v-list-item-group
          active-class="deep-purple--text text--accent-4"
        >
          <v-list-item
            v-for="(challenge, i) in challenges"
            :key="i"
            @click="navigateToChallenge(i)"
          >
            <v-list-item-title>
              {{ `${i+1} - ${challenge.name}` }}
            </v-list-item-title>
          </v-list-item>

        </v-list-item-group>
      </v-list>
    </v-navigation-drawer>
  </div>
</template>

<script>
import { mapState, mapActions } from 'vuex'

export default {
  data: () => {
    return {
      drawer: true
    }
  },
  computed: {
    ...mapState(['challenges', 'currentIndex'])
  },
  methods: {
    ...mapActions(['navigateToChallenge'])
  }
}
</script>

<style lang="sass" scoped>
// $pink: #ff0080
$orange: #fd8f50
$med-pink: darken($orange, 5%)
$dark-pink: darken($orange, 10%)
.logo-text
  font-family: 'Monoton', cursive
  color: $orange
  text-shadow: 0 0 3px $med-pink, 0 0 5px $med-pink, 0 0 8px $med-pink, 0 0 10px $dark-pink, 0 0 18px $dark-pink
.v-toolbar__title
  font-size: 2rem
.theme--dark.v-card.v-card--outlined
  border: 2px solid white
</style>
