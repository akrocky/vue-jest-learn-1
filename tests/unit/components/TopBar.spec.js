import { mount, shallowMount } from '@vue/test-utils'

import TopBar from '@/components/TopBar.vue'

describe('TopBar.vue', () => {
  test('snapshot', () => {
    const $store = {
      state: {
        challenges: [],
        currentIndex: 0
      },
      getters: {
        percentDone: () => 0
      }
    }

    const wrapper = shallowMount(TopBar, {
      propsData: {

      },
      mocks: {
        $store
      }
    })
    expect(wrapper).toMatchSnapshot()
  })
})
