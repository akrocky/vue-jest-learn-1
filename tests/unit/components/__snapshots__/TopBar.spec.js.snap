// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar.vue snapshot 1`] = `
<div>
  <v-app-bar-stub dark="true" tag="header" tile="true" extensionheight="48" src="" value="true">
    <v-toolbar-title-stub class="logo-text">
      Regex Breakfast
    </v-toolbar-title-stub>
    <v-spacer-stub></v-spacer-stub>
    <v-card-stub loaderheight="4" tag="div" outlined="true" rounded="">
      <v-card-text-stub class="white--text pa-2">
        <v-progress-linear-stub color="#ec1a25" value="() => 0" active="true" buffervalue="100" height="10"></v-progress-linear-stub>
        Challenge #1
        out of 0
      </v-card-text-stub>
    </v-card-stub>
    <v-spacer-stub></v-spacer-stub>
    <v-btn-stub tag="button" tile="true" activeclass="" outlined="true" type="button">
      Challenges
    </v-btn-stub>
  </v-app-bar-stub>
  <v-navigation-drawer-stub absolute="true" overlayopacity="0" height="100%" minivariantwidth="56" mobilebreakpoint="1264" right="true" src="" tag="aside" temporary="true" width="256" value="true">
    <v-list-stub tag="div" tile="true" dense="true" nav="true">
      <v-list-item-group-stub activeclass="deep-purple--text text--accent-4"></v-list-item-group-stub>
    </v-list-stub>
  </v-navigation-drawer-stub>
</div>
`;
